ZeCamp/ZeCamp/Sources/Schedule/ScheduleScreen.swift

```diff
@@ -86,24 +88,108 @@ struct TimeSlot {
     weak var viewController: UIViewController?
+    var timeSlots: [TimeSlot]
+    
+    init(_ timeSlots: [TimeSlot]){
+        self.timeSlots = timeSlots
+    }
     
     func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
-        let emptyScreen = UIViewController()
-        emptyScreen.title = "Event"
-        emptyScreen.view.backgroundColor = .white
+        defer {
+            tableView.deselectRow(at: indexPath, animated: true)
+        }
+        
+        
+        let id = timeSlots[indexPath.section].events[indexPath.row].id
+        print(id)
+        let eventScreen = EventScreen(id).getController();
+        
+        eventScreen.view.backgroundColor = .white
+        if #available(iOS 11.0, *) {
+            eventScreen.navigationItem.largeTitleDisplayMode = .never
+        }
+        viewController?.navigationController?.pushViewController(eventScreen, animated: true)
+    }
+    
+}
+
+class EventScreen {
+    
+    var eventId = 0
+    
+    init(_ id: Int) {
+        self.eventId = id
+    }
+    
+    func setupLabelConstraints(_ label: UILabel, _ eventScreen: UIViewController){
+        label.translatesAutoresizingMaskIntoConstraints = false
+        label.centerXAnchor.constraint(greaterThanOrEqualTo: eventScreen.view.centerXAnchor).isActive = true
+        label.widthAnchor.constraint(equalTo: eventScreen.view.widthAnchor, constant: -20).isActive = true
+        label.numberOfLines = 0
+    }
+    
+    func setDefaultStackView(_ stackView: UIStackView) {
+        stackView.axis = .vertical
+        stackView.alignment = .leading
+        stackView.distribution = .equalSpacing
+        stackView.isLayoutMarginsRelativeArrangement = true
+        stackView.layoutMargins = .init(top: 10, left: 10, bottom: 10, right: 10)
+        stackView.spacing = 10
+        stackView.translatesAutoresizingMaskIntoConstraints = false
+    }
+    
+    func constrainStackView(StackView stackView: UIStackView, ViewController eventScreen: UIViewController) {
+        let container = eventScreen.view!
+        
         if #available(iOS 11.0, *) {
-            emptyScreen.navigationItem.largeTitleDisplayMode = .never
+            container.addSubview(stackView)
+            stackView.topAnchor.constraint(equalTo: container.safeAreaLayoutGuide.topAnchor).isActive = true
+            stackView.trailingAnchor.constraint(equalTo: container.safeAreaLayoutGuide.trailingAnchor).isActive = true
+            stackView.leftAnchor.constraint(equalTo: container.safeAreaLayoutGuide.leftAnchor).isActive = true
+            stackView.rightAnchor.constraint(equalTo: container.safeAreaLayoutGuide.rightAnchor).isActive = true
+        } else {
+            // Fallback on earlier versions
         }
-        viewController?.navigationController?.pushViewController(emptyScreen, animated: true)
-        tableView.deselectRow(at: indexPath, animated: true)
     }
     
+    func getController( ) -> UIViewController {
```

Comment:
We need to add a `Screen` protocol at some point. Could you add it and make `EventScreen` and `ScheduleScreen` implement it with a method that creates a `UIViewController`?

Reply:
:+1: